src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -958,34 +957,97 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         winrt::hstring currentCommandline,
         winrt::hstring currentWorkingDirectory)
     {
+        // enumerate all the parent directories we want to import snippets from
+        std::filesystem::path directory{ std::wstring_view{ currentWorkingDirectory } };
+        std::vector<std::filesystem::path> directories;
+        while (!directory.empty())
         {
-            // Check if there are any cached commands in this directory.
+            directories.push_back(directory);
+            const auto parentPath = directory.parent_path();

```

Comment:
You _can_ remove the string copy, but it looks ugly so maybe don't.

```suggestion
            auto parentPath = directory.parent_path();
            const auto done = directory == parentPath;
            directories.push_back(std::move(directory));
            if (done)
            {
                break;
            }
            directory = std::move(parentPath);
```